File: android/vault-access/src/test/kotlin/org/secretary/browse/PurgeNoticeTest.kt
package org.secretary.browse

import kotlin.test.Test
import kotlin.test.assertEquals

class PurgeNoticeTest {
    @Test fun singlePurge() =
        assertEquals(PurgeNotice("Deleted forever", PurgeSeverity.SUCCESS),
            formatPurgeNotice(PurgeOutcome.SinglePurge))

    @Test fun emptyTrashSingular() =
        assertEquals(PurgeNotice("Purged 1 item", PurgeSeverity.SUCCESS),
            formatPurgeNotice(PurgeOutcome.EmptyTrash(purgedCount = 1, filesFailed = 0)))

    @Test fun emptyTrashPlural() =
        assertEquals(PurgeNotice("Purged 4 items", PurgeSeverity.SUCCESS),
            formatPurgeNotice(PurgeOutcome.EmptyTrash(purgedCount = 4, filesFailed = 0)))

    @Test fun oneFailedFileWarnsSingular() =
        assertEquals(PurgeNotice("Purged 4 items · 1 file could not be removed", PurgeSeverity.WARNING),
            formatPurgeNotice(PurgeOutcome.EmptyTrash(purgedCount = 4, filesFailed = 1)))

    @Test fun failedFilesWarnPlural() =
        assertEquals(PurgeNotice("Purged 4 items · 2 files could not be removed", PurgeSeverity.WARNING),
            formatPurgeNotice(PurgeOutcome.Retention(purgedCount = 4, filesFailed = 2)))

    @Test fun retentionNoop() =
        assertEquals(PurgeNotice("No items were past the retention window", PurgeSeverity.SUCCESS),
            formatPurgeNotice(PurgeOutcome.Retention(purgedCount = 0, filesFailed = 0)))

    @Test fun emptyTrashNoop() =
        assertEquals(PurgeNotice("Trash was already empty", PurgeSeverity.SUCCESS),
            formatPurgeNotice(PurgeOutcome.EmptyTrash(purgedCount = 0, filesFailed = 0)))
}

File: desktop/tests/purgeNotice.test.ts
// desktop/tests/purgeNotice.test.ts
// Unit tests for the pure post-op purge-notice formatter (#411). Pins every
// row of the design's message table incl. pluralization boundaries and the
// filesFailed warning branch. No I/O; no ambient state.
import { describe, it, expect } from 'vitest';
import { formatPurgeNotice } from '../src/lib/purgeNotice';

describe('formatPurgeNotice', () => {
  it('single-block purge is always "Deleted forever" (success)', () => {
    expect(formatPurgeNotice({ op: 'singlePurge' })).toEqual({
      text: 'Deleted forever',
      severity: 'success'
    });
  });

  it('empty-trash of one block is singular', () => {
    expect(formatPurgeNotice({ op: 'emptyTrash', purgedCount: 1, filesFailed: 0 })).toEqual({
      text: 'Purged 1 item',
      severity: 'success'
    });
  });

  it('empty-trash of several blocks is plural, success', () => {
    expect(formatPurgeNotice({ op: 'emptyTrash', purgedCount: 4, filesFailed: 0 })).toEqual({
      text: 'Purged 4 items',
      severity: 'success'
    });
  });

  it('one failed file is a singular warning', () => {
    expect(formatPurgeNotice({ op: 'emptyTrash', purgedCount: 4, filesFailed: 1 })).toEqual({
      text: 'Purged 4 items · 1 file could not be removed',
      severity: 'warning'
    });
  });

  it('multiple failed files are a plural warning', () => {
    expect(formatPurgeNotice({ op: 'retention', purgedCount: 4, filesFailed: 2 })).toEqual({
      text: 'Purged 4 items · 2 files could not be removed',
      severity: 'warning'
    });
  });

  it('retention with nothing expired is a distinct success message', () => {
    expect(formatPurgeNotice({ op: 'retention', purgedCount: 0, filesFailed: 0 })).toEqual({
      text: 'No items were past the retention window',
      severity: 'success'
    });
  });

  it('empty-trash that purged nothing (concurrent empty) reports so', () => {
    expect(formatPurgeNotice({ op: 'emptyTrash', purgedCount: 0, filesFailed: 0 })).toEqual({
      text: 'Trash was already empty',
      severity: 'success'
    });
  });
});
